### PR TITLE
SEP38: Add `buy_asset` field to GET /prices endpoint

### PR DESCRIPTION
**Pull request by lijamie98**
Add the `buy_asset` and `buy_amount` fields to the `GET /prices` endpoint.

### CODE REVIEW

File: ecosystem/sep-0038.md [hunk 1]
```diff
@@ -230,9 +230,11 @@ Optional [`SEP-10`]. If authentication is provided it can be used to personalize
...
+| `buy_asset`            | string | The asset you want to buy, using the [Asset Identification Format](#asset-identification-format). If the `buy_asset` is defined, the `sell_asset` must not exist.                                                                                                                                          |
+| `sell_amount`          | string | The amount of `sell_asset` the client would exchange for each of the `buy_assets`. The `sell_amount` is valid only when the `sell_asset` is defined.                                                                                                                                                       |
+| `buy_amount`           | string | The amount of `buy_asset` the client would exchange for each of the `sell_assets`. The `buy_amount is valid only when the `buy_asset` is defined.                                                                                                                                                          |     |
```
Comment (JakeUrban) [inline ecosystem/sep-0038.md:237]:
```suggestion
| ---------------------- | ------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --- |
| `sell_asset`           | string | The asset you want to sell, using the [Asset Identification Format](#asset-identification-format). If the `sell_asset` is defined, the `buy_asset` must not exist.                                                                                                                                         |
| `buy_asset`            | string | The asset you want to buy, using the [Asset Identification Format](#asset-identification-format). If the `buy_asset` is defined, the `sell_asset` must not exist.                                                                                                                                          |
| `sell_amount`          | string | The amount of `sell_asset` the client would exchange for each of the `buy_assets`. The `sell_amount` is valid only when the `sell_asset` is defined.                                                                                                                                                       |
| `buy_amount`           | string | The amount of `buy_asset` the client would exchange for each of the `sell_assets`. The buy_amount is valid only when the `buy_asset` is defined.                                                                                                                                                          |     |
```

File: ecosystem/sep-0038.md [hunk 2]
```diff
@@ -283,7 +292,27 @@ GET /prices?sell_asset=iso4217:BRL&sell_amount=500&country_code=BRA&sell_deliver
...
+  ]
+}
+```
+
+---
+
+Get the amount and asset needed for buying `100` of the `stellar:USDC::GA5...ZVN"`.
+
+```
+GET /prices?buy_asset=stellar:USDC:GA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN&sell_amount=100
```
Comment (JakeUrban) [inline ecosystem/sep-0038.md:306]:
```suggestion
GET /prices?buy_asset=stellar:USDC:GA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN&buy_amount=100
```

File: ecosystem/sep-0038.md [hunk 3]
```diff
@@ -7,9 +7,9 @@ Author: Jake Urban <@jakeurban> and Leigh McCulloch <@leighmcculloch>
 Track: Standard
 Status: Draft
 Created: 2021-04-09
-Updated: 2023-08-24
+Updated: 2024-08-15
 Discussion: https://github.com/stellar/stellar-protocol/issues/901
-Version 2.2.0
+Version 2.2.1
```
Comment (JakeUrban) [inline ecosystem/sep-0038.md:12]: I think this should be a minor version update since its additive, right?
```suggestion
Version 2.3.0
```

**Review by JakeUrban — APPROVED** [review]

**Review by Ifropc — COMMENTED** [review]
LGTM except that Jake raised

File: ecosystem/sep-0038.md (hunk 3 above)
Comment (lijamie98) [reply ecosystem/sep-0038.md:12]: Yes.

File: ecosystem/sep-0038.md (hunk 1 above)
Comment (lijamie98) [reply ecosystem/sep-0038.md:237]: Fixed.

File: ecosystem/sep-0038.md (hunk 2 above)
Comment (lijamie98) [reply ecosystem/sep-0038.md:306]: Fixed.

**Review by JakeUrban — APPROVED** [review]

**Review by Ifropc — APPROVED** [review]

**Review by JakeUrban — APPROVED** [review]

**Pull request merged**